### ISSUE TITLE
Train very slow on a RTX 4090

### ISSUE THREAD

**Issue by Ederzak** [VectorSpaceLab/OmniGen2#74]
Hello, i don't understand why but the Lora training are very slow. I use wsl2 all the data is on the Linux folder. I don't use "/mtn/"

07/02/2025 22:45:06 - INFO - __main__ - ***** Running training *****
07/02/2025 22:45:06 - INFO - __main__ -   Num examples = 271
07/02/2025 22:45:06 - INFO - __main__ -   Num batches each epoch = 271
07/02/2025 22:45:06 - INFO - __main__ -   Num Epochs = 23
07/02/2025 22:45:06 - INFO - __main__ -   Instantaneous batch size per device = 1
07/02/2025 22:45:06 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 2
07/02/2025 22:45:06 - INFO - __main__ -   Gradient Accumulation steps = 2
07/02/2025 22:45:06 - INFO - __main__ -   Total optimization steps = 3000
Checkpoint 'latest' does not exist. Starting a new training run.
07/02/2025 22:45:06 - INFO - __main__ -
07/02/2025 22:45:06 - INFO - __main__ - Steps:   0%|          | 0/3000 [00:00<?, ?it/s]
07/02/2025 22:45:06 - INFO - __main__ - ***** Wandb log dir: /home/sequoia/OmniGen2/wandb/offline-run-20250702_224505-vhzen91v/files *****
07/02/2025 22:49:32 - INFO - __main__ -
07/02/2025 22:49:32 - INFO - __main__ - Steps:   0%|          | 0/3000 [04:26<?, ?it/s, loss=0.396, loss-bin2-10=0.396, lr=1e-18, total_loss=0.396]
07/02/2025 22:49:32 - INFO - __main__ -
07/02/2025 22:49:32 - INFO - __main__ - Steps:   0%|          | 1/3000 [04:26<221:54:49, 266.39s/it, loss=0.396, loss-bin2-10=0.396, lr=1e-18, total_loss=0.396]
07/02/2025 22:52:09 - INFO - __main__ -
07/02/2025 22:52:09 - INFO - __main__ - Steps:   0%|          | 1/3000 [07:03<221:54:49, 266.39s/it, loss=0.553, loss-bin6-10=0.553, lr=1.6e-9, total_loss=0.553]
07/02/2025 22:52:09 - INFO - __main__ -
07/02/2025 22:52:09 - INFO - __main__ - Steps:   0%|          | 2/3000 [07:03<168:11:27, 201.96s/it, loss=0.553, loss-bin6-10=0.553, lr=1.6e-9, total_loss=0.553]
07/02/2025 22:54:47 - INFO - __main__ -
07/02/2025 22:54:47 - INFO - __main__ - Steps:   0%|          | 2/3000 [09:40<168:11:27, 201.96s/it, loss=0.459, loss-bin4-10=0.459, lr=3.2e-9, total_loss=0.459]
07/02/2025 22:54:47 - INFO - __main__ -
07/02/2025 22:54:47 - INFO - __main__ - Steps:   0%|          | 3/3000 [09:40<151:18:47, 181.76s/it, loss=0.459, loss-bin4-10=0.459, lr=3.2e-9, total_loss=0.459]
07/02/2025 22:57:22 - INFO - __main__ -
07/02/2025 22:57:22 - INFO - __main__ - Steps:   0%|          | 3/3000 [12:15<151:18:47, 181.76s/it, loss=0.306, loss-bin2-10=0.306, lr=4.8e-9, total_loss=0.306]
07/02/2025 22:57:22 - INFO - __main__ -
07/02/2025 22:57:22 - INFO - __main__ - Steps:   0%|          | 4/3000 [12:15<142:28:54, 171.21s/it, loss=0.306, loss-bin2-10=0.306, lr=4.8e-9, total_loss=0.306]
07/02/2025 22:59:56 - INFO - __main__ -
07/02/2025 22:59:56 - INFO - __main__ - Steps:   0%|          | 4/3000 [14:50<142:28:54, 171.21s/it, loss=0.519, loss-bin5-10=0.519, lr=6.4e-9, total_loss=0.519]
07/02/2025 22:59:56 - INFO - __main__ -
07/02/2025 22:59:56 - INFO - __main__ - Steps:   0%|          | 5/3000 [14:50<137:28:59, 165.26s/it, loss=0.519, loss-bin5-10=0.519, lr=6.4e-9, total_loss=0.519]
07/02/2025 23:02:31 - INFO - __main__ -
07/02/2025 23:02:31 - INFO - __main__ - Steps:   0%|          | 5/3000 [17:25<137:28:59, 165.26s/it, loss=0.346, loss-bin2-10=0.346, lr=8e-9, total_loss=0.346]
07/02/2025 23:02:31 - INFO - __main__ -
07/02/2025 23:02:31 - INFO - __main__ - Steps:   0%|          | 6/3000 [17:25<134:34:56, 161.82s/it, loss=0.346, loss-bin2-10=0.346, lr=8e-9, total_loss=0.346]
07/02/2025 23:05:06 - INFO - __main__ -
07/02/2025 23:05:06 - INFO - __main__ - Steps:   0%|          | 6/3000 [20:00<134:34:56, 161.82s/it, loss=0.388, loss-bin4-10=0.388, lr=9.6e-9, total_loss=0.388]
07/02/2025 23:05:06 - INFO - __main__ -
07/02/2025 23:05:06 - INFO - __main__ - Steps:   0%|          | 7/3000 [20:00<132:36:14, 159.50s/it, loss=0.388, loss-bin4-10=0.388, lr=9.6e-9, total_loss=0.388]
07/02/2025 23:07:41 - INFO - __main__ -
07/02/2025 23:07:41 - INFO - __main__ - Steps:   0%|          | 7/3000 [22:35<132:36:14, 159.50s/it, loss=0.329, loss-bin2-10=0.329, lr=1.12e-8, total_loss=0.329]
07/02/2025 23:07:41 - INFO - __main__ -
07/02/2025 23:07:41 - INFO - __main__ - Steps:   0%|          | 8/3000 [22:35<131:15:54, 157.94s/it, loss=0.329, loss-bin2-10=0.329, lr=1.12e-8, total_loss=0.329]
07/02/2025 23:10:15 - INFO - __main__ -
07/02/2025 23:10:15 - INFO - __main__ - Steps:   0%|          | 8/3000 [25:09<131:15:54, 157.94s/it, loss=0.419, loss-bin3-10=0.419, lr=1.28e-8, total_loss=0.419]
07/02/2025 23:10:15 - INFO - __main__ -
07/02/2025 23:10:15 - INFO - __main__ - Steps:   0%|          | 9/3000 [25:09<130:19:07, 156.85s/it, loss=0.419, loss-bin3-10=0.419, lr=1.28e-8, total_loss=0.419]
07/02/2025 23:12:50 - INFO - __main__ -
07/02/2025 23:12:50 - INFO - __main__ - Steps:   0%|          | 9/3000 [27:44<130:19:07, 156.85s/it, loss=0.535, loss-bin6-10=0.535, lr=1.44e-8, total_loss=0.535]
07/02/2025 23:12:50 - INFO - __main__ -
07/02/2025 23:12:50 - INFO - __main__ - Steps:   0%|          | 10/3000 [27:44<129:38:56, 156.10s/it, loss=0.535, loss-bin6-10=0.535, lr=1.44e-8, total_loss=0.535]
07/02/2025 23:15:24 - INFO - __main__ -
07/02/2025 23:15:24 - INFO - __main__ - Steps:   0%|          | 10/3000 [30:18<129:38:56, 156.10s/it, loss=0.389, loss-bin4-10=0.389, lr=1.6e-8, total_loss=0.389]
07/02/2025 23:15:24 - INFO - __main__ -
07/02/2025 23:15:24 - INFO - __main__ - Steps:   0%|          | 11/3000 [30:18<129:15:10, 155.67s/it, loss=0.389, loss-bin4-10=0.389, lr=1.6e-8, total_loss=0.389]
07/02/2025 23:17:59 - INFO - __main__ -
07/02/2025 23:17:59 - INFO - __main__ - Steps:   0%|          | 11/3000 [32:53<129:15:10, 155.67s/it, loss=0.719, loss-bin8-10=0.719, lr=1.76e-8, total_loss=0.719]
07/02/2025 23:17:59 - INFO - __main__ -
07/02/2025 23:17:59 - INFO - __main__ - Steps:   0%|          | 12/3000 [32:53<128:56:06, 155.34s/it, loss=0.719, loss-bin8-10=0.719, lr=1.76e-8, total_loss=0.719]
07/02/2025 23:20:34 - INFO - __main__ -
07/02/2025 23:20:34 - INFO - __main__ - Steps:   0%|          | 12/3000 [35:28<128:56:06, 155.34s/it, loss=0.523, loss-bin5-10=0.523, lr=1.92e-8, total_loss=0.523]
07/02/2025 23:20:34 - INFO - __main__ -
07/02/2025 23:20:34 - INFO - __main__ - Steps:   0%|          | 13/3000 [35:28<128:44:02, 155.15s/it, loss=0.523, loss-bin5-10=0.523, lr=1.92e-8, total_loss=0.523]
07/02/2025 23:23:08 - INFO - __main__ -
07/02/2025 23:23:08 - INFO - __main__ - Steps:   0%|          | 13/3000 [38:02<128:44:02, 155.15s/it, loss=0.565, loss-bin4-10=0.565, lr=2.08e-8, total_loss=0.565]
07/02/2025 23:23:08 - INFO - __main__ -
07/02/2025 23:23:08 - INFO - __main__ - Steps:   0%|          | 14/3000 [38:02<128:28:40, 154.90s/it, loss=0.565, loss-bin4-10=0.565, lr=2.08e-8, total_loss=0.565]
07/02/2025 23:26:13 - INFO - __main__ -
07/02/2025 23:26:13 - INFO - __main__ - Steps:   0%|          | 14/3000 [41:07<128:28:40, 154.90s/it, loss=0.472, loss-bin4-10=0.472, lr=2.24e-8, total_loss=0.472]
07/02/2025 23:26:13 - INFO - __main__ -
07/02/2025 23:26:13 - INFO - __main__ - Steps:   0%|          | 15/3000 [41:07<135:55:03, 163.92s/it, loss=0.472, loss-bin4-10=0.472, lr=2.24e-8, total_loss=0.472]
07/02/2025 23:29:19 - INFO - __main__ -
07/02/2025 23:29:19 - INFO - __main__ - Steps:   0%|          | 15/3000 [44:13<135:55:03, 163.92s/it, loss=0.515, loss-bin5-10=0.515, lr=2.4e-8, total_loss=0.515]
07/02/2025 23:29:19 - INFO - __main__ -
07/02/2025 23:29:19 - INFO - __main__ - Steps:   1%|          | 16/3000 [44:13<141:27:44, 170.66s/it, loss=0.515, loss-bin5-10=0.515, lr=2.4e-8, total_loss=0.515]

----------------

name: ft_lora

seed: 2233
device_specific_seed: true
workder_specific_seed: true

data:
  data_path: data_configs/van_gogh/van_gogh.yml
  use_chat_template: true
  maximum_text_tokens: 888
  prompt_dropout_prob: !!float 0.0001
  ref_img_dropout_prob: !!float 0.5

  # Résolution 768x768
  max_output_pixels: 589824       # 768 x 768
  max_input_pixels: [589824, 589824, 589824, 589824]
  max_side_length: 768

model:
  pretrained_vae_model_name_or_path: black-forest-labs/FLUX.1-dev
  pretrained_text_encoder_model_name_or_path: Qwen/Qwen2.5-VL-3B-Instruct
  pretrained_model_path: OmniGen2/OmniGen2

  arch_opt:
    patch_size: 2
    in_channels: 16
    hidden_size: 2520
    num_layers: 32
    num_refiner_layers: 2
    num_attention_heads: 21
    num_kv_heads: 7
    multiple_of: 256
    norm_eps: !!float 1e-05
    axes_dim_rope: [40, 40, 40]
    axes_lens: [10000, 10000, 10000]
    text_feat_dim: 2048
    timestep_scale: !!float 1000

transport:
  snr_type: lognorm
  do_shift: true
  dynamic_time_shift: true

train:
  # Dataloader
  global_batch_size: 2
  batch_size: 1
  gradient_accumulation_steps: 2

  max_train_steps: 3000           # Pour 250 images, ~12 epochs
  dataloader_num_workers: 8

  # Optimizer
  learning_rate: !!float 8e-7
  scale_lr: false
  lr_scheduler: timm_constant_with_warmup
  warmup_t: 500
  warmup_lr_init: 1e-18
  warmup_prefix: true
  t_in_epochs: false

  use_8bit_adam: true
  adam_beta1: 0.9
  adam_beta2: 0.95
  adam_weight_decay: !!float 0.01
  adam_epsilon: !!float 1e-08
  max_grad_norm: 1

  gradient_checkpointing: true

  set_grads_to_none: true

  # Precision et perfs
  allow_tf32: true               # Accélère un peu sans perte
  mixed_precision: 'bf16'        # Parfait pour RTX 4090

  ema_decay: 0.0

  # LoRA
  lora_ft: true
  lora_rank: 8
  lora_dropout: 0

val:
  validation_steps: 500
  train_visualization_steps: 1

logger:
  log_with: [wandb]
  checkpointing_steps: 250
  checkpoints_total_limit: ~

cache_dir: 
resume_from_checkpoint: latest

please help me I don't know what to do...

**Comment by sdbds** [comment]
If you have a 4090 with 24GB, then it's obviously OOMing so it's particularly slow...
This project isn't optimized, and I think it might be faster if latent are cached to disk in advance.
